design-proposals/instancetype-vm-rollout-support.md

```diff
@@ -0,0 +1,63 @@
+# Overview
+
+The [VM Rollout Strategy](TODO) feature introduced in KubeVirt 1.2 now allows VM owners to have changes made to the VM object propagates to the VMI where possible. When enabled this currently allows an owner to add additional vCPUs and memory to the VM. This design proposal covers extending this support to also allow for an owner to replace an existing instance type with a larger one causing for example additional vCPUs and memory to be propagated to the VMI.
+
+## Motivation
+
+The existing behaviour of requiring a restart for instance type and preference changes to propagate to a VMI is in-line with most attributes of a VM. However given the work already completed as part of the VM Rollout Strategy it would be useful to bring this behaviour in line with that feature when an instance type only increases the provided number of vCPUs or amount of memory.
+
+## Goals
+
+- When possible propagate changes provided by a new instance type to a VMI using the same approach as the underlying VM Rollout Strategy feature.
+
+## Non Goals
+
+- Support for the attributes of an instance type and preference not already supported by the VM Rollout Strategy feature are not covered by this proposal
+
+## Definition of Users
+
+A VM owner
+
+## User Stories
+
+- As a VM Owner I want additional resources introduced when switching to a different instance type to be propagated to the VMI where possible without a restart.
```

Comment:
I wouldn't mention a restart. How can we guarantee that it will not be required after the change?
Perhaps we can just drop the `without a restart` part?